### PR TITLE
Better Jupyter integration

### LINKED ISSUE #1070: JupyterHub and JupyterLab integration
We should be able to kick off a LiberTEM GUI session from JupyterHub and JupyterLab.

- [ ] Infrastructure on LiberTEM side
- [ ] Configuration files / instructions / documentation for integration
- [ ] If applicable, upstream sth to JupyterHub / JupyterLab / other platforms where settings can be downloaded from to make set-up easy for users.

Builds on preliminary work from @sk1p : https://github.com/sk1p/libertem-jupyter-proxy

### PR DESCRIPTION
**Pull request by sk1p**
Refs #1070 

Parts: 

- [x] make sure we properly run behind the authentication proxy
- [x] Add documentation on how a jupyterhub deployment works
- [x] Currently, a restriction is that you can't shutdown the LiberTEM server from the GUI. Or at least, if you do, it won't start properly afterwards, until you restart your notebook server via the jupyter control panel. This should be documented.
- [x] token auth for security on multi-user systems (on the LiberTEM side)
- [x] token auth (on the proxy side) requires https://github.com/LiberTEM/LiberTEM-jupyter-proxy/pull/1  and https://github.com/jupyterhub/jupyter-server-proxy/pull/287

## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes
* [N/A] I have added/updated test cases
* [x] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if changes were made to the GUI)

## Reviewer Checklist:

* [x] `/azp run libertem.libertem-data` passed

### CODE REVIEW

**Comment by uellue** [comment]
This works amazing! The item about not being able to restart should be solvable, right? After that it would be ready to roll! :-)

**Comment by sk1p** [comment]
> This works amazing! The item about not being able to restart should be solvable, right? After that it would be ready to roll! :-)

Yes, solvable, currently blocked by https://github.com/jupyterhub/jupyter-server-proxy/pull/215 - for our test deployment, we could use the PR branch, but for the documentation, I'd like to at least document installation from the official repository, if not from a release of `jupyter-server-proxy`

**Comment by sk1p** [comment]
@uellue this should now be ready for review - I'm still chasing a but though that causes our own deployment to hang when connecting, but other than that, it is ready to be merged

**Comment by sk1p** [comment]
> I'm still chasing a but though that causes our own deployment to hang when connecting

This is now fixed, too - was just an installation issue

**Comment by uellue** [comment]
Refs #67 -- only missing part is interactive use now. :-)

File: docs/source/jupyter.rst
Comment (sk1p) [inline docs/source/jupyter.rst:25]:
```suggestion
        (jupyter-venv) $ cd jupyter-server-proxy && git checkout websocket-fix-plus-restart
```

**Review by uellue — APPROVED** [review]
Awesome! 🚀  Confirmed to work on Linux, including shutdown and restart. Windows issues documented now. I LOVE the LiberTEM button in Jupyter Hub, this will be my preferred method from now on. :-)

LGTM, merging after CI.

**Pull request merged**
